File: FragmentsApp/app/src/main/java/com/rjhartsoftware/fragmentsapp/PopupFragment.java
package com.rjhartsoftware.fragmentsapp;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;


/**
 * A simple {@link Fragment} subclass.
 */
public class PopupFragment extends DialogFragment {

    public static final String TAG = "_popup";

    public PopupFragment() {
        // Required empty public constructor
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("Popup Fragment");
        builder.setMessage("This is a popup Fragment");
        builder.setPositiveButton("OK", null);
        return builder.create();
    }
}

File: FragmentsApp/app/src/main/java/com/rjhartsoftware/fragmentsapp/PopupFragment.kt
package com.rjhartsoftware.fragmentsapp

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment

/**
 * A simple [Fragment] subclass.
 */
class PopupFragment : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        builder.setTitle("Popup Fragment")
        builder.setMessage("This is a popup Fragment")
        builder.setPositiveButton("OK", null)
        return builder.create()
    }

    companion object {
        const val TAG = "_popup"
    }
}